### PR TITLE
[Feature][transform] Add a module to set default value for null field

### PR DESCRIPTION
**Pull request by Interest1-wyt**
https://github.com/apache/incubator-seatunnel/issues/1963

## Purpose of this pull request

Set default value for null field

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-etl/src/main/scala/org/apache/seatunnel/spark/transform/service/EtlHandler.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.apache.seatunnel.spark.transform.service
+
```
Comment (wuchunfu) [inline seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-etl/src/main/scala/org/apache/seatunnel/spark/transform/service/EtlHandler.scala]: @Interest1-wyt Please add the license header, thanks

**Comment by ruanwenjun** [comment]
Please add an issue and describe why you need to add this transform plugin. In fact, I am confusion with the plugin name `etl`.

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-etl/src/main/scala/org/apache/seatunnel/spark/transform/service/EtlHandler.scala (hunk 1 above)
Comment (Interest1-wyt) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-etl/src/main/scala/org/apache/seatunnel/spark/transform/service/EtlHandler.scala]: ok i'll add it later :)

**Comment by Interest1-wyt** [comment]
>I wanted to conceive an ETL plugin that contains various functions at first. The null field processing is only a small part of it. After your question, I found that the function of the module I defined was too general, and each plugin of setunnel must have its own Clear functionality, so I changed my code and module name, which now only provides null field handling. Also I have opened an issue for it.


After I commit new code,Dead Link Checker failed,It doesn't seem to be my problem. Can you help me rerun  the ci

File: docs/en/transform/nulltf.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+If there is no "=value" after the field name, the default value will be set according to the field type.
+
+## Examples
+
+the configuration
+
+```bash
+  nulltf {
+      fields = ["name","price=0","num=100","flag","dt_timestamp=2022-05-18 13:51:40.603","dt_date=2022-05-19"]
+  }
```
Comment (ruanwenjun) [inline docs/en/transform/nulltf.md]:
```suggestion
      fields : {
          price:0
          }
  }
```
Can you use the conf style here, then you don't need to parse "=".

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: docs/en/transform/nulltf.md (hunk 2 above)
Comment (Interest1-wyt) [reply docs/en/transform/nulltf.md]: Yes, I will resubmit later.
Thanks for your suggestion :)

**Review by ruanwenjun — DISMISSED** [review]
Generally looks good to me, It might be better if you can submit a PR to add e2e case for this plugin.

**Comment by ruanwenjun** [comment]
Please fix the scala checkstyle problem.

**Comment by Interest1-wyt** [comment]
OK, I will revise it and submit it later :)

**Comment by Interest1-wyt** [comment]
>

**Comment by Interest1-wyt** [comment]
Sorry, I accidentally clicked the close button, could you please help to restart the build :)

**Comment by Interest1-wyt** [comment]
> Generally looks good to me, It might be better if you can submit a PR to add e2e case for this plugin.
I try to add a test case to the e2e module, but the error "Could not find a valid Docker environment" is displayed when I run it. Running other test methods in e2e also reports this error. Why is this and how can I fix it?
The project runs on the win10 operating system and the docker component is installed.

error

**Comment by Interest1-wyt** [comment]
> > Generally looks good to me, It might be better if you can submit a PR to add e2e case for this plugin.
> > I try to add a test case to the e2e module, but the error "Could not find a valid Docker environment" is displayed when I run it. Running other test methods in e2e also reports this error. Why is this and how can I fix it?
> > The project runs on the win10 operating system and the docker component is installed.
> 
> error

Sorry, this is a problem with my local docker installation, which has been solved.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**